bottlecap/src/lifecycle/invocation_context.rs

```diff
@@ -0,0 +1,73 @@
+use std::collections::VecDeque;
+
+use tracing::debug;
+
+pub struct InvocationContext {
+    pub request_id: String,
+    pub runtime_duration_ms: f64,
+}
+
+#[allow(clippy::module_name_repetitions)]
+pub struct InvocationContextBuffer {
+    buffer: VecDeque<InvocationContext>,
+}
+
+impl Default for InvocationContextBuffer {
+    fn default() -> Self {
+        InvocationContextBuffer {
+            buffer: VecDeque::<InvocationContext>::with_capacity(5),
+        }
+    }
+}
+
+impl InvocationContextBuffer {
+    pub fn insert(&mut self, invocation_context: InvocationContext) {
+        if self.buffer.len() == self.buffer.capacity() {
```

Comment:
any reason why using a growable ring buffer?
This seems https://docs.rs/ringbuffer/latest/ringbuffer/struct.ConstGenericRingBuffer.html to do the same stuff in a simpler way
> // Because capacity is reached the next push will be the first item of the buffer.
and with stack allocation it should be more efficient, right?